qiskit/primitives/containers/observables_array.py

```diff
@@ -181,6 +198,11 @@ def ravel(self) -> ObservablesArray:
         """
         return self.reshape(self.size)
 
+    @property
+    def num_qubits(self) -> int:
```

Comment:
This line implies that the number of qubits cannot be `None`, but `None` seems like a valid result of running the constructor (when there .size is 0).

Can we set the num_qubits to 0 in the else case, assumig it's possible to get there?